### PR TITLE
Add pending delete cluster API

### PR DESCRIPTION
**Pull request by tzvatot**
@oriAdler @tbrisker @zgalor PTAL

### CODE REVIEW

**Comment by tzvatot** [comment]
Related to SDA-8434

File: model/clusters_mgmt/v1/pending_delete_cluster_type.model [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Represents a pending delete entry for a specific cluster
+class PendingDeleteCluster {
+	// Cluster is the details of the cluster that is pending deletion
```
Comment (zgalor) [inline model/clusters_mgmt/v1/pending_delete_cluster_type.model:19]: also here

File: model/clusters_mgmt/v1/pending_delete_clusters_resource.model [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+		// Order criteria.
+		//
+		// The syntax of this parameter is similar to the syntax of the _order by_ clause of
+		// a SQL statement, but using the names of the attributes of the cluster instead of
+		// the names of the columns of a table. For example, in order to sort the clusters
+		// descending by region identifier the value should be:
+		//
+		// ```sql
+		// region.id desc
```
Comment (zgalor) [inline model/clusters_mgmt/v1/pending_delete_clusters_resource.model:50]: better change to an example relevant to the pending delete cluster

**Comment by zgalor** [comment]
LGTM

**Pull request merged**